app/team.go

```diff
@@ -160,6 +161,101 @@ func (a *App) SoftDeleteAllTeamsExcept(teamID string) *model.AppError {
 	return nil
 }
 
+func shouldCreateOnboardingLinkedBoard(a *App, c request.CTX, teamId string) bool {
+	ffEnabled := a.Config().FeatureFlags.OnboardingAutoShowLinkedBoard
+	if !ffEnabled {
+		return false
+	}
+
+	const preferenceName = "linked_board_created"
+	data, sysValErr := a.Srv().Store().System().GetByName(model.PreferenceOnboarding + "_" + preferenceName)
+	if sysValErr != nil {
+		var nfErr *store.ErrNotFound
+		if errors.As(sysValErr, &nfErr) { // if no board has been registered, it can create one for this team
+			return true
+		}
+		mlog.Error("Cannot get the system values", mlog.Err(sysValErr))
+		return false
+	}
+
+	// get the team list and check if the team value has been already stored, if so, no need to create a board in town square in that team
+	teamsList := strings.Split(data.Value, ",")
+	for _, team := range teamsList {
+		if team == teamId {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (a *App) createOnboardingLinkedBoard(c request.CTX, teamId string) (*fb_model.Board, *model.AppError) {
+	const defaultChannelName = "town-square"
+	const defaultTemplatesTeam = "0"
+	const defaultTemplateTitle = "Welcome to Boards!"
+	const preferenceName = "linked_board_created"
+	userId := c.Session().UserId
+
+	boardService := a.Srv().services[product.BoardsKey].(product.BoardsService)
```

Comment:
do not use the suggetion directly
```suggestion
	boardServiceItf, ok := a.Srv().services[product.BoardsKey]
	if !ok {
	   // product.BoardsKey key not found
	}
	boardService, typeOk := boardServiceItf.(product.BoardsService)
	if !typeOk {
	   // boardServiceItf is NOT of type product.BoardsService
	}
```